### ISSUE TITLE
Accessibility: unable to open/view automation details via the keyboard

### ISSUE THREAD

**Issue by CityDweller** [home-assistant/frontend#21028]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

I am a screen reader user and only navigate the UI via the keyboard.

When I go to Setting / Automatin, I am unable to open an existing automation with the enter key when I land on it in the table.

### Describe the behavior you expected

I expect to be able to open/edit an existing automation when I hit the enter key after bringing focus to the entry.

### Steps to reproduce the issue

1. go to Settings
2. select automation & scenes
3. to to an entry in the table via the tab or arrow keys (if using a screen reader)
4. hit enter on an entry that has focus

### What version of Home Assistant Core has the issue?

2024.06

### In which browser are you experiencing the issue with?

Microsoft Edge Version 125.0.2535.85

### Which operating system are you using to run this browser?

Windows 11

**Comment by CityDweller** [comment]
Thanks Steve for the response. This use to work, I would guess before the new table style was introduced, so I'd consider this a regression.

I'm not sure I understand what you mean by browse mode. I have no issue with reading the individual cells of the table, I just cannot open the automation to work on it.

**Comment by steverep** [comment]
I am referring to "browse mode" in the NVDA sense (as opposed to focus mode).  In JAWS this would be the virtual cursor.

If you say it worked before, than perhaps I'm misunderstanding the issue.  What screen reader are you using and could you elaborate on the steps?  Specifically, what has focus and where is your virtual/browse cursor when you hit enter?

If I'm on column 2 with the automation name and hit enter with NVDA, it navigates to edit the automation as expected.

**Comment by CityDweller** [comment]
Ok, I do use JAWS and am using the Virtual Cursor.

For JAWS, I have Smart Navigation mode set to "Tables and controls" so I can use the normal arrow keys to navigate a table.

When I am in the table and my VPC is focus on the name of the automation, if I hit enter or spacebar, the state is changed to on/off.

Step by step:
in the Automation view that shows the table,
hit T to take you to the first row. The first row for me reads blank in column1, name in column2, and last activated in column 3.
hit down arrow to go to the first automation,
arrow right to get on the name column
hit enter to open the automation for editing.

**Comment by steverep** [comment]
I don't have JAWS installed to test.  Does it work as you'd expect without smart navigation enabled?

**Comment by CityDweller** [comment]
No, I did specifically try that as I was typing my earlier reply.

As a side note, if I go to Settings / Devices and Services and then the Devices tab--which shows another table of all my devices, I can go to the device column of a device and hit enter here. it will open the device details.

**Comment by steverep** [comment]
Okay, I was going to say it sounds like JAWS might be trying to be too smart and click the first control in the row, but I guess that's not it.  The devices table working as expected for you should narrow down what is happening here, so that's helpful.  I'll try to take a closer look this week sometime.

You stated you were using Edge.  Can you tell me how Chrome and/or Firefox behaves for you?

**Comment by CityDweller** [comment]
Chrome is doing the same thing; I don't have Firefox installed.

**Comment by steverep** [comment]
I suspect this is just screwed up ARIA table setup.  It used to be fine but I'm now finding NVDA doesn't navigate properly in other ways in many tables.  I haven't changed my production home config in a while so I didn't notice until now.

**Comment by karwosts** [comment]
@steverep how do you feel about adding `Edit Automation` to the overflow menu? 

I'm understanding that screen reader users have some way to 'click' the row using some custom function of the screen reader, but perhaps it might be nice if it could also be activated with just keyboard (in the absense of doing this total data table redesign that you proposed).

**Comment by steverep** [comment]
> @steverep how do you feel about adding `Edit Automation` to the overflow menu?

IMO, that would just add clutter to the menu and ultimately be several more keystrokes to accomplish the most common task.  Always better to make the thing that everyone uses accessible rather than adding more things that everyone would see (and thus probably 👎🏻  from a UX standpoint).

> I'm understanding that screen reader users have some way to 'click' the row using some custom function of the screen reader, but perhaps it might be nice if it could also be activated with just keyboard (in the absense of doing this total data table redesign that you proposed).

It was/is already keyboard accessible.  Something went wacky with the ARIA markup I think so now JAWS has this problem, and I'm noticing other problems with NVDA.  The redesign I'm talking about isn't necessary to fix this - just need to figure out what went wacky in the a11y tree.

**Comment by CityDweller** [comment]
Unfortunately, this is still an issue with the latest version.

**Comment by CityDweller** [comment]
This is still an issue for me on Edge and Firefox.

**Comment by karwosts** [comment]
Just want to mention that I tried with NVDA and did not have any trouble opening an automation from the table with the keyboard. 

- Press `T` for the table
- `Ctrl+Alt+Down` to visit a row
- `Enter` to open the automation menu.
